src/conv-core/convcore.C

```diff
@@ -1632,12 +1663,22 @@ void CsdBeginIdle(void)
 #else
   CpvAccess(cmiMyPeIdle) = 1;
 #endif // CMK_SMP
+#if CMK_ERROR_CHECKING
+  CpvAccess(idleBeginWalltime) = CmiWallTimer();
```

Comment:
CcdCallBacks(), called above, itself calls CmiWallTimer(), as does CcdRaiseCondition(), called below. Both of these currently return void, but we could change them to return the time as a double, and remove the need to have new/extra calls to CmiWallTimer()

Reply:
Of course, those times will be slightly stale by the time they are passed out of those routines, but it won't matter for this use case, and we can warn potential users of that in comments

Reply:
That's a good point. In the latest patch, I've made CcdRaiseCondition return curWallTime and used it to set idleBeginWallTime.